Zend/tests/final_properties/access/property_incrementing_decrementing_by_ref.phpt

```diff
@@ -0,0 +1,80 @@
+--TEST--
+Test that final properties can't be incremented/decremented by reference
+--FILE--
+<?php
+
+class Foo
+{
+    final public int $property1;
+    final public int $property2 = 2;
+
+    public function __construct()
+    {
+        $this->property1 = 1;
+    }
+}
+
+$foo = new Foo();
+foreach ($foo as &$property) {
+    try {
+        ++$property;
+    } catch (Error $exception) {
+        echo $exception->getMessage() . "\n";
+    }
+}
+
+var_dump($foo->property1);
+var_dump($foo->property2);
+
+$foo = new Foo();
+foreach ($foo as &$property) {
+    try {
+        $property++;
+    } catch (Error $exception) {
+        echo $exception->getMessage() . "\n";
+    }
+}
+
+var_dump($foo->property1);
+var_dump($foo->property2);
+
+foreach ($foo as &$property) {
+    try {
+        --$property;
+    } catch (Error $exception) {
+        echo $exception->getMessage() . "\n";
+    }
+}
+
+var_dump($foo->property1);
+var_dump($foo->property2);
+
+foreach ($foo as &$property) {
+    try {
+        $property--;
+    } catch (Error $exception) {
+        echo $exception->getMessage() . "\n";
+    }
+}
+
+var_dump($foo->property1);
+var_dump($foo->property2);
+
+?>
+--EXPECT--
+Cannot increment a reference held by final property Foo::$property1
```

Comment:
That means it was possible to acquire a reference to it in the first place...

Reply:
Uh, yes.. However, It's only because I couldn't prevent references in case of `foreach ($foo as &$property) {}`, while other ways of acquiring references doesn't seem to work anymore.

Reply:
Ah, I see that it was a question in your main comment. I simply didn't find find out how to deal with foreach-by-ref. Can you tell me where I should be searching?

Reply:
You'll want to insert a check somewhere around here: https://github.com/php/php-src/blob/191451da90b6fe81641d55bf782b551c65e4fd70/Zend/zend_vm_def.h#L6516